sklearn/model_selection/_search.py

```diff
@@ -513,84 +517,143 @@ def _fit(self, X, y, labels, parameter_iterable):
+        # Get all the unique parameters across all the candidates
+        parameter_names = sorted(set(chain(*map(six.iterkeys,
+                                                candidate_params))))
 
         base_estimator = clone(self.estimator)
-
         pre_dispatch = self.pre_dispatch
 
         out = Parallel(
             n_jobs=self.n_jobs, verbose=self.verbose,
             pre_dispatch=pre_dispatch
         )(delayed(_fit_and_score)(clone(base_estimator), X, y, self.scorer_,
                                   train, test, self.verbose, parameters,
-                                  self.fit_params, return_parameters=True,
+                                  self.fit_params, return_parameters=False,
                                   error_score=self.error_score)
-          for parameters in parameter_iterable
+          for parameters in candidate_params
           for train, test in cv.split(X, y, labels))
 
-        # Out is a list of triplet: score, estimator, n_test_samples
-        n_fits = len(out)
+        n_candidates = len(candidate_params)
 
-        scores = list()
-        grid_scores = list()
-        for grid_start in range(0, n_fits, n_splits):
-            n_test_samples = 0
-            score = 0
-            all_scores = []
-            for this_score, this_n_test_samples, _, parameters in \
-                    out[grid_start:grid_start + n_splits]:
-                all_scores.append(this_score)
-                if self.iid:
-                    this_score *= this_n_test_samples
-                    n_test_samples += this_n_test_samples
-                score += this_score
-            if self.iid:
-                score /= float(n_test_samples)
-            else:
-                score /= float(n_splits)
-            scores.append((score, parameters))
-            # TODO: shall we also store the test_fold_sizes?
-            grid_scores.append(_CVScoreTuple(
-                parameters,
-                score,
-                np.array(all_scores)))
-        # Store the computed scores
-        self.grid_scores_ = grid_scores
+        all_test_scores, all_test_sample_counts, _ = zip(*out)
+
+        all_test_scores = np.array(
+            all_test_scores, dtype=np.float64).reshape(n_candidates, n_splits)
+        # NOTE test_sample counts (weights) remain the same for all candidates
+        test_sample_counts = np.array(all_test_sample_counts[:n_splits],
+                                      dtype=np.int)
+
+        scorer_name = _get_scorer_name(self.scorer_)
+        results = dict()
+
+        for split_i in range(n_splits):
+            results["%s_test_split_%d" % (scorer_name, split_i)] = (
+                all_test_scores[:, split_i])
+
+        # Computed the (weighted) mean and std for all the candidates
+        mean_w = np.average(
+            all_test_scores, axis=1,
+            weights=test_sample_counts if self.iid else None)
+
+        std_w = np.sqrt(np.average(
```

Comment:
`_w` is for weighted. and `all_` is to denote that it is for all the `n_candidates`... I'll remove it... It doesn't add much meaning, now that we are vectorizing a single 2d score matrix (unlike before).